plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/EclipseProjectCreator.java

```diff
@@ -0,0 +1,21 @@
+package com.google.cloud.tools.eclipse.appengine.newproject;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+
+/**
+* Utility to make a new Eclipse project with the App Engine Standard Nature in the workspace.  
+*/
+class EclipseProjectCreator {
+
+  /**
+   * @return true on successful project creation; false otherwise
+   */
+  static boolean makeNewProject(AppEngineStandardProjectConfig config) {
+    IWorkspaceRoot workspaceRoot = ResourcesPlugin.getWorkspace().getRoot();
+    IProject project = workspaceRoot.getProject(config.getEclipseProjectName());
```

Comment:
For the actual creation process, take a look at
- `org.eclipse.ui.wizards.newresource.BasicNewProjectResourceWizard#createNewProject()` for an overview of how the creation is structured
- `org.eclipse.swt.common.project.facet.ui`, particularly the `BasicFacetedProjectWizard` because we'll have to configure the appropriate facets.
- An article on using [Progress Monitors](https://eclipse.org/articles/Article-Progress-Monitors/article.html)